connectd/connectd.c

```diff
@@ -875,12 +872,6 @@ static void try_connect_one_addr(struct connecting *connect)
 				addr = &connect->addrs[connect->addrnum];
 			}
 			freeaddrinfo(ais);
-#endif
-			tal_append_fmt(&connect->errors,
-				       "%s: EXPERIMENTAL_FEATURES needed. ",
-				       type_to_string(tmpctx,
-						      struct wireaddr_internal,
-						      addr));
 			goto next;
```

Comment:
Can someone else also confirm that `goto next;` is correct here? Or do we need a `break;` ?
I'm a bit confused right now :D

Reply:
This `#endif` looks like it was originally done a bit incorrectly; I'd expect there to be an `if` statement around the `experimental features needed` error?

That aside, `goto next` is used when you want to skip the current address info attempt without trying to connect to it.
Based on the fact that the code in the experimental block (if successful) adds a new address to try to the list of addresses, going to the next seems appropriate to me. (Go try the next address which we've now added to the list of addresses to try)

Reply:
Tanks that confirms my thoughts. Will do a rebase on new master...